src/transforms/regex_parser.rs

```diff
@@ -276,10 +273,10 @@ impl Transform for RegexParser {
                 }
 
                 for (name, value) in captures {
-                    let name = match &self.target_field {
-                        Some(target) => Atom::from(format!("{}.{}", target, name)),
-                        None => name.clone(),
-                    };
+                    let name = (&self.target_field)
+                        .as_ref()
```

Comment:
Do the compiler need both of the references (`&` vs `.as_ref()`)? I think you should only need one or the other.

Reply:
I thought so too, but it was complaining. I'll take another look.

Reply:
This is the compiler error without `as_ref()`:

```
    |
276 |                     let name = (&self.target_field)
    |                                ^^^^^^^^^^^^^^^^^^^^
    |                                |
    |                                move occurs because value has type `std::option::Option<string_cache::atom::Atom<string_cache::atom::EmptyStaticAtomSet>>`, which does not implement the `Copy` trait
    |                                help: consider borrowing the `Option`'s content: `(&self.target_field).as_ref()`
```

I'm imagining that there is probably a more idiomatic way to do it though?

Reply:
My comment was not so much that the `.as_ref()` was bad, but that you don't need both that and `&`, so: `let name = self.target_field.as_ref()…`

Reply:
Aha! I see. That does work. I think I follow. `as_ref()` already turns `self.target_field` into a reference to create the `Option<&T>`.

Reply:
Just to make sure this is 100% clear (you're not exactly wrong), the declaration for `Option::as_ref` is:
```rust
fn as_ref(&self) -> Option<&T>;
```
This means it turns a borrowed (reference to) option (`&Option<T>`) into an owned option containing a borrowed object (`Option<&T>`), both with the same lifetime as the original option. The `map` method takes the option by value (and so takes ownership) which causes the original error because the function doesn't actually own `self` or its fields.

Reply:
:+1: thanks for expounding. Knowing the specific terminology is useful.